.github/workflows/lint-python.yaml

```diff
@@ -0,0 +1,36 @@
+name: Python Lint
+
+concurrency:
+  group:  ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
Let's only run the linter when there are changes in Python files. See https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#onpushpull_requestpull_request_targetpathspaths-ignore